### DISCUSSION TITLE
Expanding the physical domain

### DISCUSSION THREAD

**Discussion by alahyari** [AMReX-Codes/amrex#2211] [Q&A] [answered]
Hello AMReX people,

I am implementing far field boundary conditions using FMM for velocity-vorticity formulation. For that, I need to confine the vorticity region within my physical domain. Thus, when the vorticity reaches the physical boundary, I have to expand the physical domain in that direction on the coarsest grid to keep the vorticity confined. Using AMReX, is it possible to do that on the fly?

Thank you in advance.

Shahab

**Comment by drummerdoc** [comment, marked as answer]
I don't know of clean tools for that, but it's a pretty simple matter to initialize a new run based on a plot file of a previous run, provided you have a strategy for initializing the state outside of the region of the earlier case.  Essentially, you could follow the MFIter loop that calls ``init_data`` with some code that reads the plot file (of the solution on the smaller domain) and overwrites-on-intersect. It's a hack, but not terribly ugly.

**Reply by alahyari** [reply]
Got it. Thank you Marc for quick reply. Doing a restart was my original plan too, but I just wondered if there is a cleaner built-in way to do this that I've missed. Thank you again.

**Reply by zingale** [reply]
with the caveat that I don't know the last time it was run, we have a utility to do this in Castro (the embiggener) that might provide some inspiration:

https://github.com/AMReX-Astro/Castro/tree/main/Util/ConvertCheckpoint

**Reply by alahyari** [reply]
Thank you, Michael. This is very useful.

**Comment by alahyari** [comment]
Another quick (and naïve) question on External BCs. Let's say I have periodic BCs on x and y planes, and external boundary functions on z.lo and z.hi, which are different functions. Based on my limited understanding, the FillPatch routines let me use a pointer function to apply external BCs that is being called for the boxes at the boundary.  What is the best approach to handle more than one external boundary function without combining them together?

Thank you again.

**Reply by WeiqunZhang** [reply]
The index in the Box is global.  So you can use that information in the function provided to amrex to figure out where the box is and call the appropriate external boundary function.

**Reply by alahyari** [reply]
Thank you, Weiqun.
